lib/iris/tests/unit/fileformats/dot/test__dot_path.py

```diff
@@ -0,0 +1,82 @@
+# (C) British Crown Copyright 2016, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for :func:`iris.fileformats.dot._dot_path`."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import os.path
+import subprocess
+
+import mock
+
+from iris.fileformats.dot import _dot_path
+
+
+class Test(tests.IrisTest):
+    def setUp(self):
+        # Because _dot_path is triggered by the initial import we
+        # reset the caching status to allow us to see what happens
+        # under different circumstances.
+        patcher = mock.patch('iris.fileformats.dot._DOT_CHECKED', new=False)
+        patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_valid_absolute_path(self):
+        # Override the configuration value for System.dot_path
+        real_path = os.path.abspath(__file__)
+        assert os.path.exists(real_path) and os.path.isabs(real_path)
+        with mock.patch('iris.config.get_option', return_value=real_path):
+            result = _dot_path()
+        self.assertEqual(result, real_path)
+
+    def test_invalid_absolute_path(self):
+        # Override the configuration value for System.dot_path
+        dummy_path = '/not_a_real_path' * 10
+        assert not os.path.exists(dummy_path)
+        with mock.patch('iris.config.get_option', return_value=dummy_path):
+            result = _dot_path()
+        self.assertIsNone(result)
+
+    def test_valid_relative_path(self):
```

Comment:
I reckon you are being too "picky". As far as the routine under test is concerned it _is_ valid relative path - it even managed to "execute it".